File: back-end/models/agent.py
from typing import Any, List, Optional
from pydantic import BaseModel


class AgentMessage(BaseModel):
    role: str  # "user" | "assistant"
    content: str


class AgentChatRequest(BaseModel):
    message: str
    history: Optional[List[AgentMessage]] = None


class AgentChatResponse(BaseModel):
    reply: str

File: front-end/src/api/agent.ts
import { apiFetch } from "./client";

export interface AgentMessage {
  role: "user" | "assistant";
  content: string;
}

export interface AgentChatRequest {
  message: string;
  history?: AgentMessage[];
}

export interface AgentChatResponse {
  reply: string;
}

export function agentChat(req: AgentChatRequest): Promise<AgentChatResponse> {
  return apiFetch("/api/v1/agent/chat", {
    method: "POST",
    body: JSON.stringify(req),
  });
}
